packages/web3-errors/src/errors/response_errors.ts

```diff
@@ -36,14 +41,19 @@ const isResponseWithError = <Error = unknown, Result = unknown>(
 const buildErrorMessage = (response: JsonRpcResponse<unknown, unknown>): string =>
 	isResponseWithError(response) ? response.error.message : '';
 
-export class ResponseError<ErrorType = unknown> extends BaseWeb3Error {
+export class ResponseError<ErrorType = unknown, RequestType = unknown> extends BaseWeb3Error {
 	public code = ERR_RESPONSE;
 	public data?: ErrorType | ErrorType[];
+	public request?: JsonRpcPayload<RequestType>;
 
-	public constructor(response: JsonRpcResponse<unknown, ErrorType>, message?: string) {
+	public constructor(
+		response: JsonRpcResponse<unknown, ErrorType>,
+		message?: string,
+		request?: JsonRpcPayload<RequestType>,
+	) {
 		super(
 			message ??
-				`Returned error: ${
+				`${
```

Comment:
I don't think prefixing every JSON response error with `Returned error: ` provides any benefit, and removing it allows errors to be identified by using `error.message` without having to parse the actual error by removing prefix every time